packages/renderer/src/lib/search/search-term-parser.ts

```
@@ -0,0 +1,83 @@
+  constructor(searchTerm: string, allowedFilters: readonly F[]) {
+    const terms: string[] = [];
+    const filters = new Map<F, string[]>();
+    const normalized = searchTerm.toLowerCase();
+
+    for (const filter of allowedFilters) {
+      filters.set(filter, []);
+    }
+
+    const regex = /(\S+:"[^"]*"|"[^"]*"|\S+)/g;
+    let match: RegExpExecArray | null;
+    while ((match = regex.exec(normalized)) !== null) {
```

Biome (2.4.10)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)